Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Driver/Version2/DocumentDriver.php

```diff
@@ -0,0 +1,79 @@
+<?php
+namespace Flowpack\ElasticSearch\ContentRepositoryAdaptor\Driver\Version2;
+
+/*
+ * This file is part of the Flowpack.ElasticSearch.ContentRepositoryAdaptor package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Flowpack\ElasticSearch\ContentRepositoryAdaptor\Driver\Version1;
+use Flowpack\ElasticSearch\ContentRepositoryAdaptor\Mapping\NodeTypeMappingBuilder;
+use Flowpack\ElasticSearch\Domain\Model\Index;
+use TYPO3\Flow\Annotations as Flow;
+use TYPO3\TYPO3CR\Domain\Model\NodeInterface;
+
+/**
+ * Fulltext Indexer Driver for Elastic version 2.x
+ *
+ * @Flow\Scope("singleton")
+ */
+class DocumentDriver extends Version1\DocumentDriver
+{
+    /**
+     * @param Index $index
+     * @param NodeInterface $node
+     * @param string $documentIdentifier
+     */
+    public function deleteByDocumentIdentifier(Index $index, NodeInterface $node, $documentIdentifier)
```

Comment:
Delete query does not exists in Elastic 2.x so we use a scroll query and sub query to delete documents